Simply filesystem helpers to check is-readable/writable/executable

azat opened the pull request:
Use one system call - faccessat(), over multiple system calls.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

azat committed:
Simply filesystem helpers to check is-readable/writable/executable

Use one system call - faccessat(), over multiple system calls.

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

davenger commented:
Failed stateless tsan 01926_order_by_desc_limit: https://github.com/ClickHouse/ClickHouse/pull/43408

davenger commented:
Integration test test_jbod_ha/test.py::test_jbod_ha failed in all runs: release, asan and tsan
It changes permissions on a dir to simulate disk failure.
@azat could the change break this test?

```
...
            # mimic disk failure
            node1.exec_in_container(
                ["bash", "-c", "chmod -R 000 /jbod1"], privileged=True, user="root"
            )
    
            time.sleep(3)
    
            # after 3 seconds jbod1 will be set as broken disk. Let's wait for another 5 seconds for data to be recovered
            time.sleep(5)
    
>           assert (
                int(
                    node1.query("select total_space from system.disks where name = 'jbod1'")
                )
                == 0
            )
E           assert 104857600 == 0
E            +  where 104857600 = int('104857600\n')
E            +    where '104857600\n' = <bound method ClickHouseInstance.query of <helpers.cluster.ClickHouseInstance object at 0x7ff8bb4a9280>>("select total_space from system.disks where name = 'jbod1'")
E            +      where <bound method ClickHouseInstance.query of <helpers.cluster.ClickHouseInstance object at 0x7ff8bb4a9280>> = <helpers.cluster.ClickHouseInstance object at 0x7ff8bb4a9280>.query
```

azat committed:
tests: fix broken disk emulation in test_jbod_ha

The problem with chmod 000 is that it is simply ignored for the owner of
the namespace (verified with kprobe for security_capable [1]),
previously it worked only cause there was a check for uid explicitly in
FS::canRead/canWrite.

  [1]: cat-10561   [001]  1340776.172944: security_capable_retprobe: (capable_wrt_inode_uidgid+0x40/0x70 <- security_capable) arg1=0xffffffff

0xffffffff is -1 and it is EPERM

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests/integration: add SYS_ADMIN capability for mount/remount

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
tests/integration: fix --privileged passthrough

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat committed:
Improve logging for broken disks

Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

azat commented:
Test failures looks unrelated now:

>Integration tests (asan) [6/6] — fail: 12, passed: 537, flaky: 1

- `test_distributed_queries_stress` - #41776

>Integration tests (release) [3/4] — fail: 1, passed: 471, flaky: 0

- `test_broken_detached_part_clean_up`

>AST fuzzer (debug) — Logical error: 'Trying to get non existing mark 18446744073709551615, while size is 0'.

- #43238
